luminex/src/org/labkey/luminex/LuminexExcelParser.java

```diff
@@ -242,6 +253,93 @@ private void parseFile() throws ExperimentException
         _parsed = true;
     }
 
+    // Keeps track of potential titrations using the following structure. 'count_descriptionM_analyteN' indicates the number of instances where descriptionM and analyteN are found on the same row.
+    // crossFilePT: {description1: {analyte1: count_description1_analyte1, analtye2: count_description1_analyte2, ...},
+    //               description2: {analyte1: count_description2_analyte1, analyte2: count_description2_analyte2, ...} ... }
+    private Map<String, HashMap<String, Integer>> buildCrossFilePTs(Map<String, HashMap<String, Integer>> crossFilePT, Map<String, Titration> potentialTitrations, String analyteName, Map<String, Integer> potentialTitrationCounts, String type)
+    {
+        for (String desc : potentialTitrations.keySet())
+        {
+            if (crossFilePT.containsKey(desc))
+            {
+                if (crossFilePT.get(desc).containsKey(analyteName))
+                {
+                    Integer count = crossFilePT.get(desc).get(analyteName);
+                    crossFilePT.get(desc).put(analyteName, count + potentialTitrationCounts.get(desc));
+                }
+                else
+                {
+                    crossFilePT.get(desc).put(analyteName, potentialTitrationCounts.get(desc));
+                }
+            }
+            else
+            {
+                HashMap<String, Integer> mini = new HashMap<>();
+
+                // Check for single-file titrations. Does not check for dilution counts.
+                if (potentialTitrationCounts.get(desc) != null)
+                {
+                    // To Reviewer: It's possible to make the runtime better by removing the description from consideration in the main function's potentialTitration(Raw or Summary)Counts
+                    // once we recognize that it is a titration. However I found this makes debugging a bit harder / perhaps the code less readable because the description count is
+                    // no longer there/updated and accurate to the data.
+                    mini.put(analyteName, potentialTitrationCounts.get(desc));
+                    crossFilePT.put(desc, mini);
+                    if ((type.contentEquals("raw")) && ((potentialTitrationCounts.get(desc)) >= LuminexDataHandler.MINIMUM_TITRATION_RAW_COUNT))

```

Comment:
I don't think I have ever used the `contentEquals()` method for string comparison before. Was there a specific reason to use it here over just using `equals()` or `equalsIgnoreCase()`?

Reply:
Nope, I've changed them to equals() for consistency.